lbrynet/core/server/DHTHashAnnouncer.py

```diff
@@ -87,12 +90,32 @@ def announce():
 
 
 class DHTHashSupplier(object):
+    # 1 hour is the min time hash will be reannounced
+    MIN_HASH_REANNOUNCE_TIME = 60*60
+    # conservative assumption of the time it takes to announce
+    # a single hash
+    SINGLE_HASH_ANNOUNCE_DURATION = 1
+
     """Classes derived from this class give hashes to a hash announcer"""
     def __init__(self, announcer):
         if announcer is not None:
             announcer.add_supplier(self)
         self.hash_announcer = announcer
-        self.hash_reannounce_time = 60 * 60  # 1 hour
 
     def hashes_to_announce(self):
         pass
+
+    """
+    Hash reannounce time is set to current time + MIN_HASH_REANNOUNCE_TIME,
+    unless we are announcing a lot of hashes at once which could cause the
+    the announce queue to pile up.  To prevent pile up, reannounce
+    only after a conservative estimate of when it will finish
+    to announce all the hashes.
+    """
+    def get_next_announce_time(self, num_hashes_to_announce=1):
+        queue_size = self.hash_announcer.hash_queue_size()+num_hashes_to_announce
+        reannounce = max(self.MIN_HASH_REANNOUNCE_TIME,
+                            queue_size*self.SINGLE_HASH_ANNOUNCE_DURATION)
+        return time.time() + reannounce
```

Comment:
Yea this patch is pretty bad in that we are increasing the dependency between BlobManager and DHTHashAnnouncer. BlobManager should not really need to handle announce time related logic at all except maybe to read and write the database entry for it. I agree that your patch would be be the direction to go next.